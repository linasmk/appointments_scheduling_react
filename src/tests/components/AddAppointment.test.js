import React from "react";
import { shallow } from "enzyme";
import {
  AddAppointment,
  AddBook,
} from "../../components/crud_components/AddAppointment";
import appointments from "../fixtures/appointments";

let onSubmit, wrapper;

beforeEach(() => {
  onSubmit = jest.fn();
  wrapper = shallow(<AddAppointment onSubmit={onSubmit} />);
});

test("Should render AddAppointment correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Should handle onSubmit", () => {
  wrapper.find("AppointmentForm").prop("onSubmit")(appointments[1]);
  expect(onSubmit).toHaveBeenLastCalledWith(appointments[1]);
});
