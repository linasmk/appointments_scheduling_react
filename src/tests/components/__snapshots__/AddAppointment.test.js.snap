// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddAppointment correctly 1`] = `
<article
  className="appointments-article"
>
  <section
    className="accordion accordion-open"
  >
    <header
      className="accordion__header"
    >
      <h2
        className="accordion__heading h-heading-styles"
      >
        Add Appointment
      </h2>
      <BsPlusCircleFill
        className="plus-circle"
        onClick={[Function]}
      />
    </header>
    <div
      className="accordion__body js-accordion-open-add"
    >
      <AppointmentForm
        onSubmit={[MockFunction]}
      />
    </div>
  </section>
</article>
`;
